File: frontend/types/delivery-center.ts
/**
 * Delivery center types matching backend schemas.
 */

export interface EmployeeApproverSummary {
  id: string;
  first_name: string;
  last_name: string;
  email: string;
}

export interface DeliveryCenter {
  id: string;
  name: string;
  code: string;
  default_currency: string;
  approvers?: EmployeeApproverSummary[];
  opportunities_count?: number;
  employees_count?: number;
}

export interface DeliveryCenterCreate {
  name: string;
  code: string;
  default_currency: string;
}

export interface DeliveryCenterUpdate {
  name?: string;
  code?: string;
  default_currency?: string;
}

export interface DeliveryCenterApproverCreate {
  employee_id: string;
}

export interface DeliveryCenterApproverResponse {
  delivery_center_id: string;
  employee_id: string;
  employee: EmployeeApproverSummary;
}

export interface DeliveryCenterApproverListResponse {
  items: DeliveryCenterApproverResponse[];
  total: number;
}

export interface DeliveryCenterListResponse {
  items: DeliveryCenter[];
  total: number;
}

export type DeliveryCenterResponse = DeliveryCenter;

File: backend/app/schemas/delivery_center.py
"""
Delivery Center Pydantic schemas for request/response validation.
"""

from pydantic import BaseModel, ConfigDict, Field
from typing import List, Optional
from uuid import UUID


class DeliveryCenterBase(BaseModel):
    """Base delivery center schema with common fields."""
    name: str = Field(..., min_length=1, max_length=50)
    code: str = Field(..., min_length=1, max_length=50)
    country_code: Optional[str] = Field(None, min_length=2, max_length=2)
    default_currency: str = Field(default="USD", min_length=3, max_length=3)


class DeliveryCenterCreate(DeliveryCenterBase):
    """Schema for creating a delivery center."""
    pass


class DeliveryCenterUpdate(BaseModel):
    """Schema for updating a delivery center."""
    name: Optional[str] = Field(None, min_length=1, max_length=50)
    code: Optional[str] = Field(None, min_length=1, max_length=50)
    country_code: Optional[str] = Field(None, min_length=2, max_length=2)
    default_currency: Optional[str] = Field(None, min_length=3, max_length=3)


class EmployeeApproverSummary(BaseModel):
    """Summary schema for employee approver."""
    id: UUID
    first_name: str
    last_name: str
    email: str
    
    class Config:
        from_attributes = True


class DeliveryCenterApproverResponse(BaseModel):
    """Schema for delivery center approver response."""
    delivery_center_id: UUID
    employee_id: UUID
    employee: EmployeeApproverSummary
    
    class Config:
        from_attributes = True


class DeliveryCenterResponse(DeliveryCenterBase):
    """Schema for delivery center response."""
    model_config = ConfigDict(from_attributes=False)

    id: UUID
    approvers: Optional[List[EmployeeApproverSummary]] = None
    opportunities_count: int = 0
    employees_count: int = 0


class DeliveryCenterListResponse(BaseModel):
    """Schema for delivery center list response."""
    items: List[DeliveryCenterResponse]
    total: int


class DeliveryCenterApproverCreate(BaseModel):
    """Schema for creating a delivery center approver."""
    employee_id: UUID


class DeliveryCenterApproverListResponse(BaseModel):
    """Schema for delivery center approver list response."""
    items: List[DeliveryCenterApproverResponse]
    total: int









